packages/core/src/compiler/shapeChecker/CheckShape.ts

```diff
@@ -0,0 +1,484 @@
+import { Result } from "true-myth";
+import { dummyIdentifier } from "../../engine/EngineUtils";
+import { Circle } from "../../shapes/Circle";
+import { Ellipse } from "../../shapes/Ellipse";
+import { Equation } from "../../shapes/Equation";
+import { Group } from "../../shapes/Group";
+import { Image } from "../../shapes/Image";
+import { Line } from "../../shapes/Line";
+import { Path } from "../../shapes/Path";
+import { Polygon } from "../../shapes/Polygon";
+import { Polyline } from "../../shapes/Polyline";
+import { Rectangle } from "../../shapes/Rectangle";
+import { Shape } from "../../shapes/Shapes";
+import { Text } from "../../shapes/Text";
+import * as ad from "../../types/ad";
+import { StyleError } from "../../types/errors";
+import { Translation } from "../../types/styleSemantics";
+import {
+  checkArrow,
+  checkCenter,
+  checkCorner,
+  checkFill,
+  checkNamed,
+  checkPoly,
+  checkProp,
+  checkRect,
+  checkRotate,
+  checkScale,
+  checkString,
+  checkStroke,
+} from "./CheckShapeHierarchyProps";
+import {
+  checkFloatV,
+  checkPathDataV,
+  checkShapeListV,
+  checkStrV,
+  checkVectorV,
+} from "./CheckValues";
+
+const { err, ok } = Result;
+
+export const checkShape = (
+  shapeType: string,
```

Comment:
Would it make sense to use `ShapeType` here instead of `string`? (genuinely just asking)

Reply:
Yes, will make that change.